src/main/java/com/google/codeu/servlets/EventListServlet.java

```diff
@@ -32,31 +32,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     List<Event> events = datastore.getAllEvents();
     if (tag != null) {
-      Iterator it = events.iterator();
-      while (it.hasNext()) {
-        Event e = (Event) it.next();
-        String description = e.getDescription();
-        String regex = "(#\\w+)";
+      // Split query into separate words using a comma as a delimiter.
+      String[] tags = tag.split(",");
+      for (int i = 0; i < tags.length; i++) {
+        String currTag = tags[i]; 
+        Iterator it = events.iterator();
+        while (it.hasNext()) {
+          Event e = (Event) it.next();
+          String description = e.getDescription();
+          String regex = "(#\\w+)";
 
-        // Retrieve all hashtags in the description
-        Pattern p = Pattern.compile(regex);
-        Matcher m = p.matcher(description);
-        boolean hashtagExists = false;
+          // Retrieve all hashtags in the description
+          Pattern p = Pattern.compile(regex);
+          Matcher m = p.matcher(description);
+          boolean hashtagExists = false;
 
-        // Iterate through all hashtag terms and only remove
-        // events that do not have the hashtag in its description.
-        while (m.find()) {
-          String hashtag = m.group(1);
-          if (hashtag.substring(1).equals(tag)) {
-            hashtagExists = true;
-            break;
+          // Iterate through all hashtag terms and only remove
+          // events that do not have the hashtag in its description.
+          while (m.find()) {
+            String hashtag = m.group(1);
+            if (hashtag.substring(1).equals(currTag)) {
```

Comment:
If you write the regex as `"#(\\w+)"`, then it will still match the same strings, but the hash symbol won't be included in the group, which will allow you to drop the `.substring(1)` call here.

Reply:
!!! Whoa! That's sick. I'll update that! Thanks :)